File: algorithms/Counting_Sort.py
# We need the time module to create some time difference between each comparison
import time

# Importing colors from colors.py
from Helper.color import *

def counting_sort(data, drawData, timeTick):
    n = max(data) + 1
    count = [0] * n
    for item in data:
        count[item] += 1
    
    k = 0
    for i in range(n):
        for j in range(count[i]):
            data[k] = i
            drawData(data, [BLUE for x in range(len(data))] )
            time.sleep(timeTick)
            k += 1

    drawData(data, [BLUE for x in range(len(data))])

File: Algorithms_Code/JavaScript/Counting_Sort.js
const Counting_Sort = (arr) => {
    const max = Math.max(...arr);
    const min = Math.min(...arr);

    const range = max - min + 1;

    const count = new Array(range);
    const output = new Array(arr.length);

    for (let i = min; i <= max; i++)
        count[i] = 0;

    for (let i = 0; i < arr.length; i++)
        count[arr[i]] += 1;

    let j = 0;

    for (let i = min; i <= max; i++) {
        while (count[i] > 0) {
            arr[j] = i;
            j++;
            count[i]--;
        }
    }
}